lib/resque/scheduler/delaying_extensions.rb

```diff
@@ -257,55 +248,49 @@ def get_last_enqueued_at(job_name)
 
       private
 
+      def decode_without_nonce(job)
+        decode(job).delete_if {|k, _| k == 'nonce'}
+      end
+
       def job_to_hash(klass, args)
-        { class: klass.to_s, args: args, queue: queue_from_class(klass) }
+        job_to_hash_with_queue(queue_from_class(klass), klass, args)
       end
 
       def job_to_hash_with_queue(queue, klass, args)
         { class: klass.to_s, args: args, queue: queue }
       end
 
+      def encode_with_nonce(hash)
+        encode(hash.merge(nonce: rand))
+      end
+
       def remove_delayed_job(encoded_job)
         return 0 if Resque.inline?
 
         timestamps = redis.smembers("timestamps:#{encoded_job}")
 
         replies = redis.pipelined do
+          redis.zrem(:delayed_queue, [encoded_job])
           timestamps.each do |key|
-            redis.lrem(key, 0, encoded_job)
             redis.srem("timestamps:#{encoded_job}", key)
           end
         end
 
         return 0 if replies.nil? || replies.empty?
-        replies.each_slice(2).map(&:first).inject(:+)
-      end
-
-      def clean_up_timestamp(key, timestamp)
-        # Use a watch here to ensure nobody adds jobs to this delayed
-        # queue while we're removing it.
-        redis.watch(key) do
-          if redis.llen(key).to_i == 0
-            # If the list is empty, remove it.
-            redis.multi do
-              redis.del(key)
-              redis.zrem(:delayed_queue_schedule, timestamp.to_i)
-            end
-          else
-            redis.redis.unwatch
-          end
-        end
+        replies.first
       end
 
       def search_first_delayed_timestamp_in_range(start_at, stop_at)
```

Comment:
`next_delayed_timestamp` is the only user of this and it's not used as mentioned in another comment, so I think this can be removed too

Reply:
Can we remove this?

Reply:
yep.